libvast/src/system/disk_monitor.cpp

```diff
@@ -101,20 +99,27 @@ disk_monitor(disk_monitor_actor::stateful_pointer<disk_monitor_state> self,
                                            self->state.dbdir, err));
       // TODO(ch20006): Add some check on the overall structure on the db dir.
       std::vector<partition_diskstate> partitions;
-      for (const auto& file : index_dir) {
-        auto partition = file.path().stem().string();
+      for (const auto& entry : index_dir) {
+        auto partition = entry.path().stem().string();
         if (partition == "index.bin")
           continue;
         uuid id;
         if (!parsers::uuid(partition, id)) {
           VAST_VERBOSE("{} failed to find partition {}", self, partition);
           continue;
         }
-        // TODO: Wrap a more generic `stat()` using `vast::path`.
-        struct stat statbuf;
-        if (::stat(file.path().c_str(), &statbuf) < 0)
-          continue;
-        partitions.push_back({id, statbuf.st_size, statbuf.st_mtime});
+
+        if (entry.is_regular_file()) {
+          std::error_code err{};
+          const auto file_size = entry.file_size(err);
+          const auto mtime = entry.last_write_time(err);
+          if (!err && file_size != static_cast<std::uintmax_t>(-1))
+            partitions.push_back({id, file_size, mtime});
```

Comment:
```suggestion
            partitions.emplace_back(id, file_size, mtime);
```

Reply:
Just changed to use `emplace_back` so we get the move construction here. 👍

Reply:
We can't do an `emplace_back` actually since the `partition_diskstate` isn't move constructible. Just changed it back to use `push_back` which will copy construct the `partition_diskstate` instances.